tests/returntypes/test_autograd_qnode_new.py

```diff
@@ -0,0 +1,1942 @@
+        assert isinstance(res[0], np.ndarray)
+
+        assert res[1].shape == (10,)
+        assert isinstance(res[1], np.ndarray)
+
+    def test_sample_combination(self, tol):
+        """Test the output of combining expval, var and sample"""
+        n_sample = 10
+
+        dev = qml.device("default.qubit", wires=3, shots=n_sample)
+
+        @qnode(dev, diff_method="parameter-shift")
+        def circuit():
+            qml.RX(0.54, wires=0)
+
+            return qml.sample(qml.PauliZ(0)), qml.expval(qml.PauliX(1)), qml.var(qml.PauliY(2))
+
+        result = circuit()
+
+        assert len(result) == 3
+        assert np.array_equal(result[0].shape, (n_sample,))
+        assert isinstance(result[1], np.ndarray)
+        assert isinstance(result[2], np.ndarray)
+        assert result[0].dtype == np.dtype("int")
+
+    def test_single_wire_sample(self, tol):
+        """Test the return type and shape of sampling a single wire"""
+        n_sample = 10
+
+        dev = qml.device("default.qubit", wires=1, shots=n_sample)
+
+        @qnode(dev)
+        def circuit():
+            qml.RX(0.54, wires=0)
+
+            return qml.sample(qml.PauliZ(0))
+
+        result = circuit()
+
+        assert isinstance(result, np.ndarray)
+        assert np.array_equal(result.shape, (n_sample,))
+
+    def test_multi_wire_sample_regular_shape(self, tol):
+        """Test the return type and shape of sampling multiple wires
+        where a rectangular array is expected"""
+        n_sample = 10
+
+        dev = qml.device("default.qubit", wires=3, shots=n_sample)
+
+        @qnode(dev)
+        def circuit():
+            return qml.sample(qml.PauliZ(0)), qml.sample(qml.PauliZ(1)), qml.sample(qml.PauliZ(2))
+
+        result = circuit()
+
+        # If all the dimensions are equal the result will end up to be a proper rectangular array
+        assert isinstance(result, tuple)
+
+        assert result[0].shape == (10,)
+        assert isinstance(result[0], np.ndarray)
+
+        assert result[1].shape == (10,)
+        assert isinstance(result[1], np.ndarray)
+
+        assert result[2].shape == (10,)
+        assert isinstance(result[2], np.ndarray)
+
+
+@pytest.mark.parametrize("dev_name,diff_method,mode", qubit_device_and_diff_method)
+class TestReturn:
+    """Class to test the shape of the Grad/Jacobian/Hessian with different return types."""
+
+    def test_execution_single_measurement_param(self, dev_name, diff_method, mode):
+        """For one measurement and one param, the gradient is a float."""
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev, interface="autograd", diff_method=diff_method)
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        a = np.array(0.1, requires_grad=True)
+
+        grad = qml.grad(circuit)(a)
+
+        if diff_method == "backprop":
+            pytest.skip("CI check differs from local results.")
```

Comment:
?

Reply:
Test are passing locally but not on the CI checks. Could you give it a try?

Reply:
Agreed that this is kind of an issue if we don't have consistent behaviour here.

Reply:
Yes here it depends on your numpy version unfortunately (since 1.23.0)